students/smitco/lesson09/mailroomOO.py

```diff
@@ -113,28 +96,33 @@ def thank_you():
                           "'Exit' will return to main menu.\n"
                           ">>") 
         donor = Donor(addressee)
-        donor.donor_name(addressee)
+        all_donors = AllDonors()
         if donor.name.lower() == "list":
-            pass
+            print(list(all_donors.donors.keys()))
         elif donor.name.lower() == "exit":
+            print("\nExiting.\n")
             flag = False
         else:
             donation = input("\nWhat is the donation amount?\n>>")
-            donor.donor_amount(addressee, donation)
-            if donor.amount in ["exit", "large", "invalid"]:
-                pass
-            else:
-                donor.donor_append(addressee, donation)
-                donor.donor_email(addressee, donation)
+            try:
+                if donation == "exit":
+                    print("\nExiting.")
+                elif int(donation) >= 10**6:
+                    print("\nThe amount entered is too large.")
+                else:
+                    donation = int(donation)
+                    donor.donor_email(addressee, donation)
+                    all_donors.donor_append(addressee, donation)
+            except ValueError:
+                print("\nInvalid entry.")
+
 
 def report():
-    all_donors = All_Donors("None", 0)
-    all_donors.donor_details()
+    all_donors = AllDonors()
     all_donors.donor_report()
 
 def letters():
-    all_donors = All_Donors("None", 0)
-    all_donors.donor_details()
+    all_donors = AllDonors()
```

Comment:
your class should only need to be instantiated once, at the start of your program.